modules/core/shared/src/main/scala-3/io/circe/Derivation.scala

```diff
@@ -133,37 +130,62 @@ private[circe] trait DerivedDecoder[A] extends DerivedInstance[A] with Decoder[A
 }
 
 private[circe] trait EncoderDerivation:
-  inline final def derived[A](using inline A: Mirror.Of[A]): Encoder.AsObject[A] =
+  inline final def derived[A](using inline A: LazyMirror[A], inline E: Encoders[A]): Encoder.AsObject[A] =
     ConfiguredEncoder.derived[A](using Configuration.default)
   inline final def derivedConfigured[A](using
-    inline A: Mirror.Of[A],
+    inline A: LazyMirror[A],
+    inline E: Encoders[A],
     inline configuration: Configuration
   ): Encoder.AsObject[A] =
     ConfiguredEncoder.derived[A]
 
-private[circe] trait EncoderDerivationRelaxed:
+private[circe] trait EncoderDerivationRelaxedLP:
+  @deprecated("Use variant that takes a LazyMirror", "0.14.8")
   inline final def derived[A: Mirror.Of](using
```

Comment:
Removing this _was_ reported as a binary incompatibility, I'm thinking because the `Mirror.Of` context bound doesn't desugar to an `inline` param?